### PR TITLE
Adding ConfigMap build inputs

### PR DESCRIPTION
**Pull request by adambkaplan**
Re-worked example to use Java/Maven application (which is amenable to both ConfigMap and Secret data).

Trello card: https://trello.com/c/RMKJxJUm/1020-5-allow-using-a-configmap-as-an-input-to-a-build-builds

### CODE REVIEW

**Comment by adambkaplan** [comment]
/hold until 3.10 branch is cut
/assign @bmcelvee 

ptal at style - when to use "config map" vs. "Config Map" vs. `ConfigMap`

**Comment by bparees** [comment]
i've added the `online` release tag but technically this won't be in online until online is update to 3.11.  Not sure if there's a more appropriate way to handle that.

File: dev_guide/builds/build_inputs.adoc
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: s/secret//

File: dev_guide/builds/build_inputs.adoc
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: looks like it's `--build-config-map`? 
https://github.com/openshift/origin/pull/19655/files#diff-924f9baa301d1772b396b1629eb2bb15R118

File: dev_guide/builds/build_inputs.adoc
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: s/with/as/

File: dev_guide/builds/build_inputs.adoc
Comment (bparees) [inline dev_guide/builds/build_inputs.adoc]: not sure why one of these is doing ADD and the other COPY, let's make them consistent.

**Review by bparees — CHANGES_REQUESTED** [review]

**Comment by adambkaplan** [comment]
[(5) Allow using a ConfigMap as an input to a build [builds]](https://trello.com/c/RMKJxJUm/1020-5-allow-using-a-configmap-as-an-input-to-a-build-builds)

File: dev_guide/builds/build_inputs.adoc
Comment (xiuwang) [inline dev_guide/builds/build_inputs.adoc]: s/configMap/configmap

File: dev_guide/builds/build_inputs.adoc
Comment (xiuwang) [inline dev_guide/builds/build_inputs.adoc]: s/--contextDir/--context-dir

File: dev_guide/builds/build_inputs.adoc
Comment (xiuwang) [inline dev_guide/builds/build_inputs.adoc]: s/--contextDir/--context-dir

File: dev_guide/builds/build_inputs.adoc
Comment (xiuwang) [inline dev_guide/builds/build_inputs.adoc]: s/config map/configmap ?

Comment (adambkaplan) [reply dev_guide/builds/build_inputs.adoc]: I'll defer to @bmcelvee on conventions around kube objects like `ConfigMap`

Comment (bmcelvee) [reply dev_guide/builds/build_inputs.adoc]: Let's go with ConfigMap.

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/config maps/ConfigMaps

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/Config Map/ConfigMaps

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/config map/ConfigMap

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/configMap/ConfigMap

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/configMap/ConfigMap

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/config maps/ConfigMaps

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/config maps/ConfigMaps

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/config maps/ConfigMaps

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/config maps/ConfigMaps

**Comment by adambkaplan** [comment]
@bmcelvee should we also /s/secret/Secret/ when we are referencing a kube Secret object?

**Comment by bmcelvee** [comment]
@adambkaplan I think it's good as is. We tend to keep it as 'secret' (the Kube docs do as well).

File: dev_guide/builds/build_inputs.adoc
Comment (bmcelvee) [inline dev_guide/builds/build_inputs.adoc]: s/an Java application via/a Java application with

**Comment by bmcelvee** [comment]
@adambkaplan overall LGTM! Just one more edit and squash commits, then we'll be ready to merge. :zap:

**Comment by adambkaplan** [comment]
@bmcelvee squashed and rebased

**Pull request merged**
